src/edu/psu/sweng500/team8/gui/BoardGUI.java

```diff
@@ -0,0 +1,261 @@
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][0], gridBagConstraints);
+
+		blocks[0][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 0;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][1], gridBagConstraints);
+
+		blocks[0][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 0;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][2], gridBagConstraints);
+
+		blocks[1][0] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 0;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 1, 0, 0);
+		add(blocks[1][0], gridBagConstraints);
+
+		blocks[1][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+		add(blocks[1][1], gridBagConstraints);
+
+		blocks[1][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 1);
+		add(blocks[1][2], gridBagConstraints);
+
+		blocks[2][0] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 0;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 1, 1, 0);
+		add(blocks[2][0], gridBagConstraints);
+
+		blocks[2][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 0);
+		add(blocks[2][1], gridBagConstraints);
+
+		blocks[2][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
+		add(blocks[2][2], gridBagConstraints);
+	}
+
+	public void populatePanel(GameSession gameSession, boolean isRefresh) {
+
+		this.gameSession = gameSession;
+		this.selectedCell = null;
+
+		blocks[0][0].populate(gameSession.getGameBoard().getBlock(0),

```

Comment:
well, I was gonna go to bed but then I saw this.

I know where you are coming from. One of CS classes showing method call
being pushed to stack along with arguments and popping, etc. I think mine
was CS 201, but it's been too long to remember.

This is actually related to a conversation I had with a colleague of mine
before. A long story short, it's is actually NOT a good habit to micro
optimize such situation. To share what I learned from the discussion.

1. JIT (though based on platform, but most likely) does the optimization.
Yes, you can trust this JIT. JIT is generous and powerful.
2. Like you said such optimization might/might not give us microseconds or
nanoseconds performance gain that will not be noticeable at all. While in
programming the dev time to actually deliver the change is more costly than
the potential gain. So while it doesn't matter in terms of performance and
even readability (IMHO), it actually hurts to make such changes (in real
world situation) due to the time spent doing so.
3. Might not be completely relevant to our case, but for other cases where
the coding is not completed, it's better to make method calls like we have
now until we are toward the end of the development cycle. When we are ready
to run a profiler or some kind of tool to detect bottle neck. If the
development is not completed, we cannot guaranty the two ways will behave
exactly the same.
4. Related to 3. it provides better maintainability. If the method being
called is updated later. We don't have to worry about the caller's variable
getting assigned to.

Optimization is great. But you don't optimize too early and you don't
micro-optimize.

Let me know if you have any questions or concerns.

.





On Sun, Jul 12, 2015 at 11:21 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/BoardGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34431262>:
>
> > +
> > +		blocks[2][2] = new BlockGUI();
> > +		gridBagConstraints = new java.awt.GridBagConstraints();
> > +		gridBagConstraints.gridx = 2;
> > +		gridBagConstraints.gridy = 2;
> > +		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
> > +//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
> > +		add(blocks[2][2], gridBagConstraints);
> > +	}
> > +
> > +	public void populatePanel(GameSession gameSession, boolean isRefresh) {
> > +
> > +		this.gameSession = gameSession;
> > +		this.selectedCell = null;
> > +
> > +		blocks[0][0].populate(gameSession.getGameBoard().getBlock(0),
>
> Assume getGameBoard is a slow call (e.g. as if it has to calculate
> something). In reality it's just returning the variable so it won't make
> much of a difference, but it's still pushing/popping the call stack
> unnecessarily. Thus it is more efficient to only do it once. We're talking
> a few nanoseconds difference, but it's a good habit to get into.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34431262>.
>